pandas/io/parsers/readers.py

```diff
@@ -1846,7 +1851,29 @@ def read(self, nrows: int | None = None) -> DataFrame:
             else:
                 new_rows = len(index)
 
-            df = DataFrame(col_dict, columns=columns, index=index)
+            if hasattr(self, "orig_options"):
+                dtype_arg = self.orig_options.get("dtype", None)
```

Comment:
Is the `dtype` option normally applied in `_engine.read`? Just curious why it needs to be done here

Reply:
Yes, but the DataFrame constructor infers object to string again if the option is set, which would discard the original dtype